a10_octavia/common/utils.py

```diff
@@ -223,25 +232,28 @@ def get_network_driver():
 def get_patched_ip_address(ip, cidr, ip_version):
     net_ip, netmask = get_net_info_from_cidr(cidr, ip_version)
     if ip_version == 6:
+        is_valid_ip = validate_partial_ipv6(ip)
         octets = ip.lstrip(':').split(':')
-        if len(octets) == 8:
+        if len(octets) == 8 and not is_valid_ip:
             validate_ipv6(ip)
             if check_ip_in_subnet_range(ip, net_ip, netmask, ip_version, cidr):
                 return ip
             else:
                 raise exceptions.VRIDIPNotInSubentRangeError(ip, cidr)
         else:
-            new_ip = str(ipaddress.IPv6Address(net_ip)) + str(ip)
-            if check_ip_in_subnet_range(new_ip, net_ip, netmask, ip_version, cidr):
-                return new_ip
+            if not is_valid_ip:
+                ip = ip.replace(':', "")
```

Comment:
consider dual stack support. Either:
- allow .3 for ipv6 subnet, or
- we have another vrid_floating_ipv6 option for ipv6

Reply:
Allowed .3 in both the cases(IPv4 and IPv6)